### PR TITLE
Add MySQL fuzzy prase packet test

### PR DESCRIPTION
**Pull request by smallinsky**
## What: 
Add fuzzy test for MySQL packet parsing.

Issue [#1494](https://github.com/gravitational/cloud/issues/1494)

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]
Good catch and fuzz is awesome.

File: lib/srv/db/mysql/protocol/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//go:build go1.18
```
Comment (r0mant) [inline lib/srv/db/mysql/protocol/fuzz_test.go:1]: I think we're not on 1.18 yet, and v9 and v8 are still (and will be) built by Go 1.17 so backports will probably require regular tests?

**Review by r0mant — APPROVED** [review]

**Comment by jakule** [comment]
@smallinsky Maybe I'm missing something, but the title of this PR is "Fix mysql parse packet" and I see only a test. How the test fixes the problem? Also is the `[]byte("00000")` problematic input?

**Comment by r0mant** [comment]
@jakule @smallinsky I swear the fix was here when I looked at it this morning. Where did it go? 😄

**Comment by greedy52** [comment]
@jakule @r0mant I've also fixed the same in my other change:
- https://github.com/gravitational/teleport/pull/11914

It was `idx < -1`, now master is
https://github.com/gravitational/teleport/blob/d3bade57386c960e11425e45af520d75dfea255c/lib/srv/db/mysql/protocol/command.go#L200-L205

I assume @smallinsky has merged the fix so now just the fuzz test left

**Comment by jakule** [comment]
@greedy52 Makes sense. I'll rebase my branches. Maybe a few of my bugs will be fixed 🤣 
@smallinsky In this case can we change the name of this PR to "add mysql test" or something similar as there is no more "the fix" in it?

**Comment by smallinsky** [comment]
The actually issue was fixed by Stave  in https://github.com/gravitational/teleport/pull/11914 where test coverage was added https://github.com/gravitational/teleport/blob/d3bade57386c960e11425e45af520d75dfea255c/lib/srv/db/mysql/protocol/packet_test.go#L373
 
 I have renamed PR to reflect the actually changes in this PR.

**Pull request merged**